visvis2/material/_triangle.py

```diff
@@ -1,7 +1,11 @@
 from ._base import Material
 
 import wgpu  # only for flags/enums
-from python_shader import python2shader, RES_INPUT, RES_OUTPUT
+from python_shader import python2shader, RES_INPUT, RES_OUTPUT, RES_UNIFORM
+from python_shader import Struct, vec3
+
+
+uniform_type = Struct(color=vec3)
```

Comment:
See how we define the uniform structure once, and then use it both in the shader and the material!